src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java

```diff
@@ -163,6 +172,21 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
         | DuplicateFlightIdException ex) {
       return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, ex);
     }
+
+    if (maxBadRecordsReached) {
+      List<String> loadErrors =
+          loadService.getFailedLoads(loadId, concurrentFiles).stream()
+              .map(lf -> lf.getSourcePath() + " -> " + lf.getTargetPath() + ": " + lf.getError())
+              .collect(Collectors.toList());
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_FATAL,
+          new IngestFailureException(
+              String.format(
+                  "More than %d files failed to ingest, which was the allowed amount."
+                      + " See error details for the first %d errors ",
+                  maxFailedFileLoads, concurrentFiles),
+              loadErrors));
+    }
```

Comment:
So, this means it will send the flight through the ingest undo steps. What will happen to files that were successfully loaded?

Reply:
They'll stay there. If the user re-submits with the same load id, the flight will pick up where it left off. If they use a different load id, it'll fail because the files are already there, unless they've chosen to resolve existing paths.